### PR TITLE
Jakarta Enterprise Beans 3.2 Specification

### PR DESCRIPTION
**Pull request by dblevins**
Signed-off-by: David Blevins <dblevins@tomitribe.com>

## Specification PR template
When creating a specification project release review, create two PRs with the content divided as follows.

Include the following in PR#1:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of wombat_x.y.pdf
- [x] Specification HTML in the form of wombat_x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [x] For a Release Review, a summary that a Compatible Implementation
      is complete, passes the TCK, and that the TCK includes sufficient
      coverage of the specification. The TCK users guide MUST include
      the instructions to run the compatible implementations used to
      validate the release.  Instructions MAY be by reference.
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      https://oss.sonatype.org/content/repositories/jakartaejb-1014/jakarta/ejb/jakarta.ejb-api/3.2.6/
      https://oss.sonatype.org/content/repositories/staging/jakarta/ejb/jakarta.ejb-api/3.2.6/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      N/A - Covered by [Jakarta EE Platform TCK](http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee8-eftl/promoted/eclipse-jakartaeetck-8.0.0.zip)
- [x] The URL of the compatibility certification request issue:
      N/A - Covered by [Eclipse Glassfish Certification Request
](https://github.com/eclipse-ee4j/jakartaee-platform/issues/99) https://github.com/eclipse-ee4j/ejb-api/issues/56

Include the following in PR#2:
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory.

### CODE REVIEW

File: enterprise-beans/3.2/_index.md [hunk 1]
```diff
@@ -1,11 +1,16 @@
 ---
 title: "Jakarta Enterprise Beans 3.2"
 date: 2019-07-19T15:57:57-0700
```
Comment (bshannon) [inline enterprise-beans/3.2/_index.md]: Time component not needed.

File: enterprise-beans/3.2/_index.md [hunk 2]
```diff
@@ -25,4 +30,4 @@ Jakarta Enterprise Beans defines an architecture for the development and deploym
 
 # Compatible Implementations
 
-* Jakarta Enterprise Beans 3.2.5
\ No newline at end of file
+* Jakarta Enterprise Beans 3.2.6
```
Comment (bshannon) [inline enterprise-beans/3.2/_index.md]: The Compatible Implementation of Enterprise Beans is GlassFish 5.1.
Also need a link.

**Comment by dblevins** [comment]
Note to self. This spec is due for review and promotion to vote August 21st.  Review this checklist and ensure all requirements are met:
https://github.com/jakartaee/specification-committee/blob/master/spec_review_checklist.md

**Comment by kwsutter** [comment]
# Spec Review Checklist

1. Spec PR
  - [ ] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form {spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form {spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://oss.sonatype.org/conte/repositories/staging/jakarta/{spec}/jakarta-{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] Second PR for just apidocs

2. _index.md
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request

**Comment by NottyCode** [comment]
# Spec Review Checklist

1. Spec PR
  - [x] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [x] Directory of form {spec}/x.y
  - [x] PDF of form {spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [x] HTML of form {spec}-spec-x.y.html ("-spec" preferred but not required)
  - [x] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [x] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [x] Staging repository link of the form https://oss.sonatype.org/conte/repositories/staging/jakarta/{spec}/jakarta-{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [x] Second PR for just apidocs

2. _index.md
  - [x] Link to spec pdf
  - [x] Link to spec html
  - [x] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip
  - [x] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation

3. javadocs
  - [x] Footer contains Eclipse copyright and link to license
  - [x] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [x] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [x] No DRAFT or SNAPSHOT
  - [x] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request

**Comment by NottyCode** [comment]
Following issues found (some are covered by Bill already, but I found a few more)

1. There is no {spec}/_index.md.
1. There is no mention of the TCK being part of the platform, a link there would be useful
1. No compatibility certification link
1. The Javadoc includes a META-INF containing a license file with CDDL license in it
1. The Javadoc jar in the staged repo does not match the PR, it doesn't contain the ESFL license
1. The version in the spec has a micro version
1. The Eclipse copyright is not in the specification
1. I can't fill out steps 6-9 due to the lack of TCK or CI.
1. The javadoc PR contains an _index.md file

**Comment by dblevins** [comment]
- Compatible implementation https://github.com/jakartaee/specifications/pull/36/files#diff-51f3180117389848d15d7e38248de84fR33
- Root _index.md file https://github.com/jakartaee/specifications/pull/36/files#diff-27c9f6d394f27c4daab588ffb33d861fR1-R4
- Two digit spec version https://github.com/jakartaee/specifications/pull/36/files#diff-dc5ec1c3f130d7331b6e7eb0d0145c8bR532
- Specification copyright https://github.com/jakartaee/specifications/pull/36/files#diff-dc5ec1c3f130d7331b6e7eb0d0145c8bR566-R570

**Comment by dblevins** [comment]
Updated to `final` as per today's Spec Committee call.

**Comment by bshannon** [comment]
@dblevins This PR should refer to a certification request in the ejb-api project.

**Comment by edbratt** [comment]
Shouldn't this point to a Compatibility Certification request to the EJB specification project? https://github.com/eclipse-ee4j/ejb-api/issues

**Comment by starksm64** [comment]
In looking at the apidocs for a recent staging url:
https://deploy-preview-112--jakartaee-specifications.netlify.com/specifications/enterprise-beans/3.2/

I'm seeing that the 3.2/_index.md is missing the content links(specs, apidocs, tck, api, ...).

**Comment by bshannon** [comment]
It's picking up the "example" enterprise-beans directory that David put out there.
That staging URL must be for some other spec; there's no staging URL for this
enterprise-beans spec because it hasn't been updated since the staging mechanism
was put in place.

**Comment by dblevins** [comment]
* On ballot completion
 - [x] The designated specification committee representative adds this final checklist to the main PR.
 - [x] The specification committee representative merges the specification PRs.
 - [x] The designated specification committee representative calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job.
 - [x] The specification project member who created the staging release promotes the specification api jars to maven central. An example release job script can be found here https://wiki.eclipse.org/MavenReleaseScript.
 - [x] The EMO updates the specification page with the ballot results.
This list goes on the committed spec index page.
 - [ ] The specification project team should go through the merged spec website page to verify all the links are valid.
 - [x] The specification project team should approve the compatibility request.
 - [ ] The compatible implementation project/vendor MUST send an email tck@eclipse.org for approval of the compatible implementation for trademark usage.
 - [ ] The specification project team should merge any final release branch as appropriate for the branch management for the project.

**Pull request merged**